lib/chai/core/assertions.js

```diff
@@ -988,9 +988,33 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+
+    var target = flag(this, 'object')
+    var validInstanceOfTarget = constructor === Object(constructor) && (
+        typeof constructor === 'function' ||
+        (typeof Symbol !== 'undefined' && typeof Symbol.hasInstance !== 'undefined' &&
+         typeof constructor[Symbol.hasInstance] === 'function')
```

Comment:
@meeber I can trigger the getter for @@hasInstance zero times if I use the `in` operator.

I also removed the `try/catch` statements on this commit.

What do you think about this?

Is there anything I should improve?